modules/module-mongodb-storage/src/storage/MongoBucketStorage.ts

```diff
@@ -84,6 +84,23 @@ export class MongoBucketStorage
     return storage;
   }
 
+  async getSystemIdentifier(): Promise<storage.BucketStorageSystemIdentifier> {
+    let id = 'N/A';
+    try {
+      const result = await this.client
+        .db('local')
+        .collection('system.replset')
+        .findOne({}, { projection: { _id: 1 } });
+      id = result?._id.toString() ?? id;
+    } catch (ex) {
+      logger.warn(`Could not query MongoDB replica set id`, ex);
+    }
```

Comment:
You can use this approach to get the replicaset name, which doesn't require any priviledges:

https://github.com/powersync-ja/powersync-service/blob/08cf53247e36089be3dbf833812a389f38b338fe/modules/module-mongodb/src/replication/replication-utils.ts#L12-L20

I'd also recommend not catching the error - depending on what we use this for it can cause more issues if we silently return `N/A` rather than throwing an error. (I'm thinking use cases like detecting when the source db was switched).

Reply:
Thanks! I've updated to use the linked approach for fetching the replica set name. 
It seems like the replica set _name_ might not have the same cluster/server wide uniqueness that the Postgres system identifier has (I believe it can be manually specified when creating a replica set). The replica set Id could be more similar, but that does not seem to be available with this method. Other commands for fetching the replica set id seem to require elevated levels of permissions. 
I suppose the replica set name is okay for now.